### PR TITLE
Allow users to accept (internal) shares by default

### LINKED ISSUE #18255: No way to accept shares if notifications app is not enabled
Since https://github.com/nextcloud/server/pull/16828, when a file is shared but the notifications app is not installed, there is no way to accept the share nor is any feedback given, everything looks fine for the person creating the share, but the recipient doesn't see anything.

Not having the notifications app is probably not common (outside git dev installs) but it's not a force enabled app

@nickvergessen @rullzer

### PR DESCRIPTION
**Pull request by rullzer**
Fixes #18255 

Todo:
- [x] Merge #18384 
- [x] Add webUI to actually set this
  * Should only be available to configure this if the config.php override is not set to true

### CODE REVIEW

File: lib/public/Share/Events/SharedEvent.php [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+namespace OCP\Share\Events;
+
+use OCP\EventDispatcher\Event;
+use OCP\Share\IShare;
+
+class SharedEvent extends Event {
```
Comment (ChristophWurst) [inline lib/public/Share/Events/SharedEvent.php]: `ShareCreatedEvent`?

**Review by ChristophWurst — COMMENTED** [review]
Looks good

event naming is not ideal IMO

**Review by skjnldsv — APPROVED** [review]

**Review by ChristophWurst — APPROVED** [review]
Looks good!

**Comment by daita** [comment]
I just had a bad UX. True that I am not used to having to confirm for a shared file but:

- there is only a (discret) red dot on the notification icon.
- the shared file is displayed in the header as 'Recently shared' but clicking on the file does nothing as the file is not available in the folder/filetree. Maybe displaying a message to accept the share when clicking on the file.
- we might add a popup in the middle of the screen saying that someone is sharing a file when a file is shared

**Comment by rullzer** [comment]
@daita please create a separate issue

**Pull request merged**